js/client-teambuilder.js

```diff
@@ -1245,6 +1254,39 @@
 			this.save();
 		},
 
+		setNotesBlur: function (e) {
+			var textarea = $(e.currentTarget);
+			if (!$.trim(textarea.val())) textarea.parent().hide();
+			if (textarea.val().indexOf('|') >= 0 || textarea.val().indexOf(']') >= 0 || textarea.val().indexOf('\t') >= 0) {
+				app.addPopupMessage("Unfortunately, set notes can't contain tabs or the characters | and ], since they're used for storing teams.");
```

Comment:
Allowing either | or ] would indeed be possible, however it would make it difficult to ever add more elements to the packed set unless yet another separator would be used for that.
Allowing both | and ] would not be possible, as far as I can tell, because then I can't know where the next set starts.